src/app/modules/product/component/product-image/product-image.component.ts

```diff
@@ -0,0 +1,213 @@
+    if(this.form.invalid) return;
+    this.submitted = false;
+
+    this.productService.updateProduct(this.form.value, this.productUpdate).subscribe({
+      next: (v) => {
+        this.getProduct();
+        this.hideModalForm();
+        this.resetVariables();
+        this.swal.successMessage(v.message);
+      },
+      error: (e) => {
+        console.log(e);
+        this.swal.errorMessage("No se pudo actualizar la categoria");
+      }
+    });
+  }
+
+  // image 
+  uploadProductImage(image: string){
+    // creamos el objeto customer image
+    let productImageC: ProductImage = new ProductImage();
+    productImageC.product_id = this.product.product_id;
+    productImageC.image = image;
+ 
+    // enviamos la imagen a la API
+    this.productImageService.uploadProductImage(productImageC).subscribe({
+      next: (v) => {
+        this.swal.successMessage(v.message);
+        this.getProduct();
+        this.getProductImages();
+      },
+      error: (e) => {
+        console.error(e);
+        this.swal.errorMessage(e.error.message);
+      }
+    });
+  }
+
+  fileChangeHandler($event: any) {
+    this.ngxService.open($event, {
+      aspectRatio: 1 / 1,
+      autoCropArea: 1,
+      resizeToWidth: 360,
+      resizeToHeight: 360,
+    }).subscribe(data => {
+      this.uploadProductImage(data.base64!);
+    });
+  }
+
+  redirect(url : String){
+    this.router.navigate([url]);
+  }
+
+  getProductImages(){
+    this.loading = true;
+    this.productImageService.getProductImages(this.product.product_id).subscribe({
+      next: (v) => {
+        console.log(v);
+        this.productImages = v;
+        this.loading= false;
+      },
+      error: (e) => {
+        console.log(e);
+        this.swal.errorMessage(e.error!.message);
+        this.loading = false;
+      }
+    });
+  } 
+  showModalForm(){
+    $("#modalForm").modal("show");
+    this.getActiveCategories();
+    this.form.reset();
+    this.submitted = false;
+  }
+
+  hideModalForm(){
+    $("#modalForm").modal("hide");
+  }
+
+  // catalogues 
+
+  getActiveCategories(){
+    this.categoryService.getActiveCategories().subscribe({
+      next: (v) => {
+        this.categories = v;
+      },
+      error: (e) => {
+        console.log(e);
+        this.swal.errorMessage(e.error!.message); // show message
+      }
+    });
+  }
+
+  resetVariables(){
+    this.form.reset();
+    this.submitted = false;
+    this.productUpdate = 0;
+  }
+
+  updateProduct(product : Product){
+    this.resetVariables();
+    this.showModalForm();
+
+    this.productUpdate = product.product_id;
+    this.form.controls['product'].setValue(product.product);
+    this.form.controls['gtin'].setValue(product.gtin);
+    this.form.controls['descripcion'].setValue(product.description);
+    this.form.controls['price'].setValue(product.price);
+    this.form.controls['stock'].setValue(product.stock);
+    this.form.controls['category_id'].setValue(product.category_id);
+  }
```

Comment:
Utilizamos el producto que ya obtuvimos del servicio
```suggestion
  updateProduct(){
    this.resetVariables();
    this.showModalForm();

    this.productUpdate = this.product.product_id;
    this.form.controls['product'].setValue(this.product.product);
    this.form.controls['gtin'].setValue(this.product.gtin);
    this.form.controls['descripcion'].setValue(this.product.description);
    this.form.controls['price'].setValue(this.product.price);
    this.form.controls['stock'].setValue(this.product.stock);
    this.form.controls['category_id'].setValue(this.product.category_id);
  }
```